lib/core/test/unit/Cardano/WalletSpec.hs

```diff
@@ -505,8 +511,25 @@ instance Arbitrary (Hash "Tx") where
         Hash . BS.pack <$> replicateM 32 arbitrary
 
 instance Arbitrary Tx where
-    shrink _ = []
-    arbitrary = return $ Tx [] []
+    shrink (Tx ins outs) =
+        [Tx ins' outs | ins' <- shrinkList' ins ] ++
+        [Tx ins outs' | outs' <- shrinkList' outs ]
+      where
+        shrinkList' xs  = filter (not . null)
+            [ take n xs | Positive n <- shrink (Positive $ length xs) ]
+    arbitrary = Tx
+        <$> fmap (L.nub . L.take 5 . getNonEmpty) arbitrary
+        <*> fmap (L.take 5 . getNonEmpty) arbitrary
+
+instance Arbitrary TxIn where
+    arbitrary = TxIn
+        <$> (Hash . B8.pack <$> vectorOf 32 arbitrary)
+        <*> scale (`mod` 3) arbitrary
+
+instance Arbitrary TxOut where
+    arbitrary = TxOut
+        <$> pure (Address "address")
+        <*> (Coin <$> choose (1, 100000))
```

Comment:
One of the test was now failing because the DBLayer now calculates the transaction id itself, however, all generated transactions were defined as `Tx [] []` and therefore, had a same id! Now, it properly generates arbitrary transactions.